packages/cli/src/lib/update-check.ts

```diff
@@ -0,0 +1,233 @@
+import { mkdir, readFile, writeFile } from 'node:fs/promises';
```

Comment:
is there an npm package we can use to already do this update check? I think a lot of projects do this (including vercel/vercel) so we don't have to reinvent the wheel on this I think?